plugins/modules/discovery_intent.py

```diff
@@ -200,6 +200,101 @@
+                type: list
+                elements: dict
+                suboptions:
+                    username:
+                        description: Username for HTTP Write authentication, mandatory when using global HTTP credentials.
+                        type: str
+                    description:
+                        description: Name of the HTTP Write credential, mandatory when using  global HTTP credentials.
+                        type: str
+            snmp_v2_read_credential_list:
+                description:
+                    - List of Global SNMP V2 Read credentials to be used during device discovery.
+                    - Generally it is advised to create device credentials with unique username or description.
+                type: list
+                elements: dict
+                suboptions:
+                    username:
+                        description: Username for SNMP Read authentication, mandatory when using global SNMP credentials.
+                        type: str
+                    description:
+                        description: Name of the SNMP Read credential, mandatory when using  global SNMP credentials.
+                        type: str
+            snmp_v2_write_credential_list:
+                description:
+                    - List of Global SNMP V2 Write credentials to be used during device discovery.
+                    - Generally it is advised to create device credentials with unique username or description.
+                type: list
+                elements: dict
+                suboptions:
+                    username:
+                        description: Username for SNMP Write authentication, mandatory when using global SNMP credentials.
+                        type: str
+                    description:
+                        description: Name of the SNMP Write credential, mandatory when using global SNMP credentials.
+                        type: str
+            snmp_v3_credential_list:
+                description:
+                    - List of Global SNMP V3 credentials to be used during device discovery, giving read and write mode.
+                    - Generally it is advised to create device credentials with unique username or description.
+                type: list
+                elements: dict
+                suboptions:
+                    username:
+                        description: Username for SNMP V3 authentication, mandatory when using global SNMP credentials.
+                        type: str
+                    description:
+                        description: Name of the SNMP V3 credential, mandatory when using global SNMP credentials.
+                        type: str
+            net_conf_port_list:
+                description:
+                    - List of Global Net conf ports to be used during device discovery.
+                    - Generally it is advised to create device credentials with unique description.
```

Comment:
It's recommended to create device credentials with unique description for easy identification